src/dataviews/dataview-model-base.js

```diff
@@ -134,6 +134,39 @@ module.exports = Model.extend({
     });
   },
 
+  fetch: function (opts) {
+    this.trigger('loading', this);
+    if (this.layer.getDataProvider()) {
+      this._fetchFromDataProvider(opts);
+    } else {
+      return Model.prototype.fetch.call(this, opts);
+    }
+  },
+
+  _fetchFromDataProvider: function (opts) {
+    var dataProvider = this.layer.getDataProvider();
+
+    // TODO: At the beginning, the dataProvider will not have any features
+    // loaded so we will have to listen to the `featuresChanged` event, but
+    // once the dataProvider has been initialized, we should implement and use
+    // dataProvider.getFeatures() instead of adding yet another binding.
+    dataProvider.bind('featuresChanged', function (features) {
```

Comment:
Yes, that's why I added the TODO. I could go ahead and fix it right now, but I think it's more important to move ahead with the filtering part.